app/routines/generate_reset_code.rb

```diff
@@ -0,0 +1,18 @@
+class GenerateResetCode
```

Comment:
I feel `generate_reset_code` and `set_password` should either both be routines or methods, but not one routine and one method.